### PR TITLE
Applying Two-way-data-binding

### PR DESCRIPTION
**Pull request by dersonsena**
As mentioned in issue #33 , it is not a good practice to manipulate or read the DOM directly for reasons of performance, productivity, and component maintenance.

React.js works with state manipulation, where it takes care of manipulating the DOM in a more performative way using Virtual DOM. See more details in the [official documentation](https://pt-br.reactjs.org/docs/state-and-lifecycle.html).

Aware of this, I did the refactoring of the `Login.js` component, removing the direct access to the form fields and creating the `email` and `password` state properties.

Finally, in the `fetchQuery` function I did the `Early Return` application by removing the `else` from the conditional block. According to the **Object Calistenics** exercises, it is good practice to use Early Return to remove `else` statements.

### CODE REVIEW

**Comment by dersonsena** [comment]
The last commits performed were to meet the requirements of the code climante. I created a component called `FormField` to abstract how the label field blocks are created.

An array of objects was also created to store the field settings so that the render function was iterated and reduce redundancy even further.

File: react/src/Login.js [hunk 1]
```diff
@@ -16,51 +17,73 @@ const query = graphql`
 class Login extends React.Component {
```
Comment (caioflores) [inline react/src/Login.js]: @dersonsena Maybe we can refactor this class component to a functional one, and use hooks to set the state, what do you think?

Comment (dersonsena) [reply react/src/Login.js]: Did you means says to use the new feature [State Hooks](https://reactjs.org/docs/hooks-state.html)? For this, I should verificate the actual react.js version, given that this feature was recently launched.

Comment (caioflores) [reply react/src/Login.js]: Exactly, we already updated our production environment to use it. I think that you will not have any compatibility problems.

Comment (dersonsena) [reply react/src/Login.js]: Ok, I'll refactor this without problems. =)

Comment (caioflores) [reply react/src/Login.js]: Great, thanks!

**Comment by dersonsena** [comment]
@caioflores I refactor the Login class to use the functional approach with State Hooks. It really was very good.

Additionally I created a `form` tag, wrap de elements for inside them and rename the `handleClick` function to `handleSubmit`

**Comment by caioflores** [comment]
It's much better! Hooks are great! Thanks for the changes, I'll approve it.

**Pull request merged**